src/plone/restapi/tests/test_documentation.py

```diff
@@ -545,15 +536,45 @@ def test_documentation_types_document_crud(self):
                 "required": True,
             },
         )
-        save_request_and_response_for_docs("types_document_post_field", response)
+        # With plone.dexterity 2.10+ we get an unstable behavior name like this:
+        # plone.dexterity.schema.generated.plone_5_1630611587_2_523689_0_Document
+        # In older versions, we got this:
+        # plone.dexterity.schema.generated.plone_0_Document
+        # Normalize this to look like the new name
+        document_schema_re = re.compile(
+            r"^plone.dexterity.schema.generated.plone_.*_Document$"
+        )
```

Comment:
I think we can be less invasive here. That is, replacing only the really dynamic part. I suggest:

```suggestion
        document_schema_re = re.compile(
            r"^plone.dexterity.schema.generated.plone_5_\d*_2_\d*_0_Document$"
        )
```
So, if we have changes like what happened in `plone.dexterity 2.10+`, the tests will fail.

Reply:
Why would we do this?
I have no idea if the 5, 2 and 0 are always here or if they can differ as well. Seems pretty random to me.
This is an internal dexterity detail. If it changes, I do *not* want these tests to fail.

Reply:
They will only change if we have structural changes like the ones that occurred not `plone.dexterity 2.10`. They don't change "via parameter". In this case, I find it valid for the test to fail, as failed. If we had this regular expression, before the change from `plone.dexterity 2.10`, the tests would not fail, even with the name pattern changing. And I think that in this case it should fail.

`schemaName` after this line:

https://github.com/plone/plone.dexterity/blob/35a12ab726b76ee5527ee467974917bc1fbdca0c/plone/dexterity/fti.py#L262

it's always something like:

`plone_5_1631640310_2_6213937_0_Document`

Reply:
I see the 5 and 0 come from the [`SchemaNameEncoder`](https://github.com/plone/plone.dexterity/blob/35a12ab726b76ee5527ee467974917bc1fbdca0c/plone/dexterity/schema.py#L295-L321).
I see no value in having a test in `plone.restapi` fail when this class changes.

Reply:
We can have more similar values to the real ones in the documentation. I think that's why we have testing to generate documentation, rather than just writing the documentation.

Reply:
Fine, since no one stepped up to argue in my favour, I have changed it: commit 8921becee228c0fa2c43f31ed5d4731119a53bc7.

I think it is fine to merge now, right? Well, once everything is green.